MobileWeather/WeatherService.m

```diff
@@ -51,87 +54,68 @@ - (void)dealloc {
 }
 
 - (void)start {
-    if ([self isStarted]) return;
+    if (self.isStarted) return;
     
     [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(handleUpdateNotification:) name:MobileWeatherLocationUpdateNotification object:nil];
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(handleUpdateNotification:) name:MobileWeatherTimeUpdateNotification object:nil];
     [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(handleUpdateNotification:) name:MobileWeatherLanguageUpdateNotification object:nil];
     
     // load api key from settings
-    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
-    [self setServiceApiKey:[defaults objectForKey:PREFS_WEATHER_API_KEY]];
+//    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+    self.serviceApiKey = PREFS_WEATHER_API_KEY;
     
```

Comment:
Should be `[self setServiceApiKey:[defaults objectForKey:PREFS_WEATHER_API_KEY]];`, otherwise, it can't get the API key entered by the user in *Settings*.

Reply:
I prefer the approach of letting the dev set it in the settings so that they don't have to re-enter it on the phone constantly (if they delete the app, if they use a different simulator or phone, etc.)

Reply:
But it should definitely be documented

Reply:
I think that works better too. The wiki for this project needs to be updated with this instruction. Also the `weatherapikey` textfield should be deleted from the **root.plist** file.